fe/fe-core/src/main/java/com/starrocks/meta/lock/LockManager.java

```diff
@@ -0,0 +1,521 @@
+                        break;
+                    }
+
+                    boolean lockTimeOut = (timeout != 0) && timeRemain(timeout, startTime) < 0;
+                    if (lockTimeOut) {
+                        removeFromWaiterList(rid, locker, lockType);
+
+                        /* Failure to acquire lock within the timeout ms*/
+                        throw new LockTimeoutException("");
+                    }
+
+                    /*
+                     * There are two reasons for the loop below.
+                     *
+                     * 1. When another thread detects a deadlock and notifies this thread,
+                     * it will wakeup before the timeout interval has expired. We must loop
+                     * again to perform deadlock detection. Normally, if the deadlock
+                     * detected by the other thread is still present, this locker will be
+                     * selected as the victim and we will throw DeadLockException below.
+                     *
+                     * 2. spurious wakeups
+                     */
+                }
+
+                if (victim == null) {
+                    assert isOwner(rid, locker, lockType);
+                    locker.clearWaitingFor();
+                }
+            }
+
+            if (victim == null) {
+                /* Locker owns the lock and no deadlock was detected. */
+                return;
+            } else {
+                /*
+                 * A deadlock is detected and this locker is not the victim.
+                 * Notify the victim.
+                 */
+                boolean currentLockerIsOwner = notifyVictim(victim, locker, rid, lockType, timeout, startTime);
+                if (currentLockerIsOwner) {
+                    synchronized (locker) {
+                        locker.clearWaitingFor();
+                    }
+                    return;
+                }
+
+                /* Retry */
+            }
+        }
+    }
+
+    private boolean notifyVictim(Locker targetedVictim, Locker currentLocker, Long rid, LockType lockType,
+                                 Long timeout, Long startTime)
+            throws LockInterruptException, DeadlockException {
+        DeadLockChecker dc = null;
+        while (true) {
+            boolean lockTimeOut = (timeout != 0) && timeRemain(timeout, startTime) < 0;
+            if (lockTimeOut && dc != null) {
+                removeFromWaiterList(rid, currentLocker, lockType);
+                /* Failure to acquire lock within the timeout ms */
+                DeadlockException exception = DeadlockException.makeDeadlockException(dc, currentLocker, false);
+                LOG.warn(exception.getMessage());
+                throw exception;
+            }
+
+            /*
+             * Notify the victim and sleep for 1ms to allow the victim to wakeup and abort.
+             */
+            synchronized (targetedVictim) {
+                targetedVictim.notify();
```

Comment:
victim is the one chosen which need to release its locks, why notify it?

Reply:
Wake up the victim, and then cause the lock operation of this Locker to throw an exception